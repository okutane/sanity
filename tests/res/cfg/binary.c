static int IR, I0, I1;

static void testInt() {
    IR = I0 + I1;
    IR = I0 - I1;
    IR = I0 * I1;
    IR = I0 / I1;
    IR = I0 % I1;
    IR = I0 & I1;
    IR = I0 | I1;
    IR = I0 ^ I1;
    IR = I0 << I1;
    IR = I0 >> I1;
}

static void testIntCmp() {
    IR = I0 < I1;
    IR = I0 > I1;
    IR = I0 <= I1;
    IR = I0 >= I1;
    IR = I0 == I1;
    IR = I0 != I1;
}