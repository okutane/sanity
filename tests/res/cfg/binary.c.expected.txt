CFG: testInt
assign: tmp0 = *I0 + *I1
binary.c:4 IR = I0 + I1;
assign: *IR = tmp0
binary.c:4 IR = I0 + I1;
assign: tmp1 = *I0 - *I1
binary.c:5 IR = I0 - I1;
assign: *IR = tmp1
binary.c:5 IR = I0 - I1;
assign: tmp2 = *I0 * *I1
binary.c:6 IR = I0 * I1;
assign: *IR = tmp2
binary.c:6 IR = I0 * I1;
assign: tmp3 = *I0 / *I1
binary.c:7 IR = I0 / I1;
assign: *IR = tmp3
binary.c:7 IR = I0 / I1;
assign: tmp4 = *I0 % *I1
binary.c:8 IR = I0 % I1;
assign: *IR = tmp4
binary.c:8 IR = I0 % I1;
assign: tmp5 = *I0 & *I1
binary.c:9 IR = I0 & I1;
assign: *IR = tmp5
binary.c:9 IR = I0 & I1;
assign: tmp6 = *I0 | *I1
binary.c:10 IR = I0 | I1;
assign: *IR = tmp6
binary.c:10 IR = I0 | I1;
assign: tmp7 = *I0 ^ *I1
binary.c:11 IR = I0 ^ I1;
assign: *IR = tmp7
binary.c:11 IR = I0 ^ I1;
assign: tmp8 = *I0 << *I1
binary.c:12 IR = I0 << I1;
assign: *IR = tmp8
binary.c:12 IR = I0 << I1;
assign: tmp9 = *I0 >> *I1
binary.c:13 IR = I0 >> I1;
assign: *IR = tmp9
binary.c:13 IR = I0 >> I1;

CFG: testIntCmp
assign: tmp0 = *I0 < *I1
binary.c:17 IR = I0 < I1;
assign: *IR = tmp0
binary.c:17 IR = I0 < I1;
assign: tmp1 = *I0 > *I1
binary.c:18 IR = I0 > I1;
assign: *IR = tmp1
binary.c:18 IR = I0 > I1;
assign: tmp2 = *I0 <= *I1
binary.c:19 IR = I0 <= I1;
assign: *IR = tmp2
binary.c:19 IR = I0 <= I1;
assign: tmp3 = *I0 >= *I1
binary.c:20 IR = I0 >= I1;
assign: *IR = tmp3
binary.c:20 IR = I0 >= I1;
assign: tmp4 = *I0 == *I1
binary.c:21 IR = I0 == I1;
assign: *IR = tmp4
binary.c:21 IR = I0 == I1;
assign: tmp5 = *I0 != *I1
binary.c:22 IR = I0 != I1;
assign: *IR = tmp5
binary.c:22 IR = I0 != I1;

